pkg/api/v1/atlasdatabaseuser_types.go

```diff
@@ -76,9 +76,32 @@ type AtlasDatabaseUserSpec struct {
 	// PasswordSecret is a reference to the Secret keeping the user password.
 	PasswordSecret *common.ResourceRef `json:"passwordSecretRef,omitempty"`
 
-	// Username is a username for authenticating to MongoDB.
+	// Username is a username for authenticating to MongoDB
+	// Human-readable label that represents the user that authenticates to MongoDB. The format of this label depends on the method of authentication:
+	// In case of AWS IAM: the value should be AWS ARN for the IAM User/Role;
+	// In case of OIDC: the value should be the Identity Provider ID;
+	// In case of Plain text auth: the value can be anything
```

Comment:
Is this actually true? I do not believe usernames allow for any arbitrary characters, they are usually letters and numbers, with some extra charts like - or _ or / allowed, but for instance, without spaces. Should we point to mongodb username format restrictions here?

Reply:
Looks like the only constraint there is the length: https://github.com/10gen/mms/blob/44ef23b3e8125a5d295be60120512d194c515474/server/src/main/com/xgen/svc/mms/api/view/atlas/ApiAtlasDatabaseUserView.java#L65-L72

Reply:
does it make sense to xref that location here as well?